website/addons/osfstorage/views.py

```diff
@@ -164,14 +164,17 @@ def osfstorage_create_child(file_node, payload, node_addon, **kwargs):
                     })
                 ),
                 dict(payload['metadata'], **payload['hashes'])
-            )
+            ).find_matching_archive()
```

Comment:
I'd prefer to avoid overriding the constructor of an MODM model.

See above.

I'm skimming off the version just because it leads to the odd situation of version not being defined in some cases. Its easy enough to throw `version = None` into the else block.